pkg/odo/genericclioptions/context.go

```diff
@@ -74,8 +81,27 @@ func newContext(command *cobra.Command, createAppIfNeeded bool) *Context {
 	} else {
 		// otherwise use the current project
 		ns = project.GetCurrent(client)
+		// if no current project, then check if user is trying to create or delete something other than project
+		if len(ns) <= 0 {
+			errFormat := "could not get current project.  Please create or set a project\n\t%s project create|set <project_name>"
+			checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command, errFormat)
+		}
+		// If 'default' project set then check if user is trying to create or delete something other than project
+		if ns == "default" {
+			errFormat := "Current project is default, which is read only.  Please create or set different project\n\t%s project create|set <project_name>"
```

Comment:
Should be `read-only`.

A bit of an odd message, `default` for my cluster isn't read only?

Reply:
`Please create or set a different project`.

Also the next section is a bit odd... It has a new-line and tab which is unconventional to our other log messages?

Reply:
> Should be `read-only`.
> 
> A bit of an odd message, `default` for my cluster isn't read only?

It is technically not read only, and i will go ahead and remove that bit, but default project is set when any user does not have a project, and OpenShift considers you not having an active project  in a standard OpenShift environment that comes as is by default. 

Its an edge case, so its simply safer to prevent user from doing things in 'default' project which:
- could potentially be manipulated by multiple users, without having knowledge about it (Tom may find his component deleted by Jerry, though he does not remember allowing Jerry access to the project, while Jerry is complaining how did he get a project that has some random component he did not ask for)

Reply:
> `Please create or set a different project`.
> 
> Also the next section is a bit odd... It has a new-line and tab which is unconventional to our other log messages?

could you please point me to a sample where a command is recommended to be run, after the error message so that I can update accordingly. Currently I am just following pattern of OpenShift messaging

Reply:
Addressed